### PR TITLE
Remove Rack::Auth::Digest

### PR DESCRIPTION
**Pull request by ninoseki**
Remove `Rack::Auth::Digest`. 

As far as I understand, #2294 reaches a conclusion that removing `Rack::Auth::Digest` is the way to go.

(It's my first time to make a PR to this repository. So please let me know if I miss something)

### CODE REVIEW

**Comment by dblock** [comment]
Let's document this in UPGRADING, please?

I think we can start following major version semver from here, and bump the version of Grape to 2.0 in this PR since this is a visibly breaking change? Could you please do it?

**Comment by ninoseki** [comment]
Sure, I updated README, CHANGELOG and also UPGRADING. Let me know if it's not enough or if I miss something.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
-### 1.9.0 (Next)
+### 2.0.0 (Next)
```
Comment (dblock) [inline CHANGELOG.md:1]: Increment the version in version.rb as well, pls.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 
 #### Features
 
 * [#2353](https://github.com/ruby-grape/grape/pull/2353): Added Rails 7.1 support - [@ericproulx](https://github.com/ericproulx).
 * [#2355](https://github.com/ruby-grape/grape/pull/2355): Set response headers based on Rack version - [@schinery](https://github.com/schinery).
 * [#2360](https://github.com/ruby-grape/grape/pull/2360): Reduce gem size by removing specs - [@ericproulx](https://github.com/ericproulx).
+* [#2361](https://github.com/ruby-grape/grape/pull/2361): Remove rack::auth::digest - [@ninoseki](https://github.com/ninoseki).
```
Comment (dblock) [inline CHANGELOG.md:8]: quote and capitalize to match the actual class name, `Rack::Auth::Digest`.

**Review by dblock — CHANGES_REQUESTED** [review]
3 more places: The README has a "Stable Version" section that will need to say next is 2.0.

File: CHANGELOG.md (hunk 1 above)
Comment (ninoseki) [reply CHANGELOG.md:1]: Incremented by [9d75263](https://github.com/ruby-grape/grape/pull/2361/commits/9d7526342a1cc7ef327f90b7fcdacc8126dcc574).

File: CHANGELOG.md (hunk 2 above)
Comment (ninoseki) [reply CHANGELOG.md]: Did it by [1310747](https://github.com/ruby-grape/grape/pull/2361/commits/1310747db21ae53d4ace475d2fde5ca97e617edd) and [3125460](https://github.com/ruby-grape/grape/pull/2361/commits/31254606d51508c44809361a98a955988620c1f9). (Sorry for dirty commit history)

**Comment by dblock** [comment]
README#stable-release still needs a bump to 2.0, please?

**Comment by ninoseki** [comment]
> README#stable-release still needs a bump to 2.0, please?

Oops, I updated it by [4905cc6](https://github.com/ruby-grape/grape/pull/2361/commits/4905cc6bb538bf770fb085247a573ed276c422f0).

**Review by dblock — APPROVED** [review]

**Pull request merged**